src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] arr;
+    private T[] newArr;
```

Comment:
Don't use Objects, Arrays, or any other util class. The 'newArr' variable should not be a class member since it's only used within methods. Declare it inside the method where it's needed.